spec/unit/cookbook/metadata_spec.rb

```diff
@@ -684,9 +788,14 @@
       metadata.attribute "bizspark/has_login",
         :display_name => "You have nothing"
       metadata.version "1.2.3"
+      metadata.chef_version ">= 11.14.2", "< 11.18.10"
+      metadata.chef_version ">= 12.2.1", "< 12.5.1"
+      metadata.ohai_version ">= 7.1.0", "< 7.5.0"
+      metadata.ohai_version ">= 8.0.1", "< 8.6.0"
     end
 
     it "should produce the same output from to_json and Chef::JSONCompat" do
+      # XXX: fairly certain this is testing ruby method dispatch
```

Comment:
Yeah, what I think you want here is to prove that you're meeting some definition of API stability that you're willing to sign up for. You can probably do that by parsing `metadata.to_json` to a Hash and then checking that to whatever level of exactness you can pull off.